frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/attachmentSelectTable.jspf

```diff
@@ -925,20 +734,54 @@
             }
         });
 
+        function hideUnhideRow(element, hide) {
+            if(hide) {
+                $(element).hide();
+            } else {
+                $(element).show();
+            }
+
+            if($(element).hasClass("expanded")) {
+                let releaseTrId = $(element).data("tt-id");
+                $("#LinkedProjectsInfo tr").each(function() {
+                    let parentTrId = $(this).data("tt-parent-id");
+                    if(parentTrId && parentTrId === releaseTrId) {
+                        if(hide) {
+                            $(this).hide();
+                        } else {
+                            $(this).show();
+                        }
+                    }
+                });
+            }
+        }
+
         $("#pills-onlyapproved-tab").on("click", function(event) {
-            if ($('#pills-onlyapproved-tab').hasClass('show active')) {
-              } else {
-                 $('#pills-showall-tab').removeClass('show active');
-                 $('#pills-onlyapproved-tab').addClass('show active');
-                 $('#pills-onlyapprovedView').addClass('show active');
-                 $('#pills-showallView').removeClass('show active');
-                 $('#LinkedProjectsInfoForOnlyApprovedAttachment').find('tr.branch').each(function(index, element) {
-                 var data = $(element).data();
-                 if (data.rowType !== 'attachment') {
-                          $('#LinkedProjectsInfoForOnlyApprovedAttachment').treetable("expandNode", data.ttId);
-                      }
+            if (!$('#pills-onlyapproved-tab').hasClass('active')) {
+                $(".sw360-tt-CheckStatus-NOTCHECKED").each(function() {
+                    hideUnhideRow($(this).parents("tr:first"), true);
+                });
+
+                $(".sw360-tt-CheckStatus-REJECTED").each(function(){
+                    hideUnhideRow($(this).parents("tr:first"), true);
+                });
+            }
+        });
+
+        $("#pills-showall-tab").on("click", function(event) {
+            if (!$('#pills-showall-tab').hasClass('active')) {
+                $(".sw360-tt-CheckStatus-NOTCHECKED").each(function() {
```

Comment:
May be use `sw360-tt-CheckStatus-REJECTED` here as well like `$(".sw360-tt-CheckStatus-NOTCHECKED, sw360-tt-CheckStatus-REJECTED")`

Reply:
Done